### ISSUE TITLE
Add Flower Baseline: FedAvg+MNIST

### ISSUE THREAD

**Issue by charlesbvll** [adap/flower#1538]
### Paper

McMahan et al., 2016, Communication-Efficient Learning of Deep Networks from Decentralized Data, MNIST

### Link

https://arxiv.org/abs/1602.05629

### Maybe give motivations about why the paper should be implemented as a baseline.

This paper was the first to propose a federated approach for deep learning on decentralized data.

### For first time contributors

- [x]  Read the [`first contribution` doc](https://flower.dev/docs/first-time-contributors.html)
- [x]  Complete the Flower tutorial
- [x]  Read the Flower Baselines docs to get an overview:
    - [x]  [https://flower.dev/docs/using-baselines.html](https://flower.dev/docs/using-baselines.html)
    - [x]  [https://flower.dev/docs/contributing-baselines.html](https://flower.dev/docs/contributing-baselines.html)

### Prepare - understand the scope

- [x] Read the paper linked above
- [x] Create the directory structure in Flower Baselines (just the `__init__.py` files and a `README.md`)
- [x] Before starting to write code, write down all of the specs of this experiment in a README (dataset, partitioning, model, number of clients, all hyperparameters, …)
- [x] Open a draft PR

### Implement - make it work

- [x] Implement some form of dataset loading and partitioning in a separate `dataset.py` (doesn’t have to match the paper exactly)
- [x] Implement the model in PyTorch
- [x] Write a test that shows that the model has the number of parameters mentioned in the paper
- [x] Implement the federated learning setup outlined in the paper, maybe starting with fewer clients
- [x] Plot accuracy and loss
- [x] Run it and check if the model starts to converge

### Align - make it converge

- [x] Implement the exact data partitioning outlined in the paper
- [x] Use the exact hyperparameters outlined in the paper
- [x] Make it converge to roughly the same accuracy that the paper states
- [x] Commit the final hyperparameters and plots
- [x] Mark the PR as ready
